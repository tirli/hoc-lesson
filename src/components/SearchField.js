import React, { Component } from 'react';

export default  class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
    }
  }

  onInputChange = (event) => {
    this.setState({
      value: event.target.value,
    });
  }
  
  onSearch = (event) => {
    event.preventDefault();
    this.props.onSearch(this.state.value);
  }

  render() {
    const { value } = this.state;

    return (
      <form>
        <input value={value} onChange={this.onInputChange} name="search" />
        <button onClick={this.onSearch} > Search </button>
      </form>
    );
  }
} 
