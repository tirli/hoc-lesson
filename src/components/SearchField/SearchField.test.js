import React from 'react';
import SearchField from './SearchField';
import { shallow, mount } from 'enzyme';

describe('SearchField', () => {
  it('shallow renders as expected', () => {
    const search = shallow(<SearchField />).html();
    expect(search).toMatchSnapshot();
  });

  it('calls handleSubmit on button click', () => {
    const mockSearch = jest.fn();
    const search = mount(<SearchField onSearch={mockSearch} />);
    search.setState({ value: 'test' })
    search.find('button').simulate('click');
    expect(mockSearch).toHaveBeenCalledTimes(1);
    expect(mockSearch).toHaveBeenCalledWith('test');
  });

  it('calls setState after input change', () => {
    const setState = jest.fn();
    const search = mount(<SearchField />);
    search.instance().setState = setState;
    search.find('input').simulate('change', { target: { value: 'test' } });
    expect(setState).toHaveBeenCalledTimes(1);
    expect(setState).toHaveBeenCalledWith({ value: 'test' });
  });
})

