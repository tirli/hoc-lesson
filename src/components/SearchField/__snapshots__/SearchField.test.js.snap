exports[`SearchField shallow renders as expected 1`] = `"<form><input value=\"\" name=\"search\" class=\"test-input\"/><button class=\"test-search-btn\"> Search </button></form>"`;
